platforms/enterprise/enterprise-operations/src/main/java/org/gradle/internal/configurationcache/ConfigurationCacheCheckFingerprintBuildOperationType.java

```diff
@@ -0,0 +1,155 @@
+     *     The {@link #getBuildInvalidationReasons()} list contains the invalidation reasons.
+     *     The {@link #getProjectInvalidationReasons()} list is empty.</li>
+     * </ul>
+     *
+     * @since 8.10
+     */
+    public interface Result {
+
+        /**
+         * Returns the overall status of the fingerprint check.
+         *
+         * @return the status
+         * @since 8.10
+         */
+        CheckStatus getStatus();
+
+        /**
+         * Returns the list of the build fingerprint invalidation reasons. Can be empty if all build-level inputs are up-to-date.
+         * The entry may still be partially invalidated if any of the project-scoped fingerprints are invalid.
+         * <p>
+         * Not all invalidation reasons may be returned. However, if the entry is invalidated, at least one reason is listed.
+         * <p>
+         * The first invalidation in the list is what Gradle shows to the user.
+         *
+         * @return the list of invalidation reasons
+         * @since 8.10
+         */
+        List<FingerprintInvalidationReason> getBuildInvalidationReasons();
+
+        /**
+         * Returns a list of project-level invalidation reasons. Only contains entries for projects that were invalidated.
+         * This list is empty if the whole build is invalidated, i.e. when {@link #getBuildInvalidationReasons()} returns a non-empty list.
+         * <p>
+         * All invalidated project have entries in this list, however, not all invalidation reasons may be listed.
+         * <p>
+         * The first invalidation of the first project in the list is what Gradle shows to the user.
+         *
+         * @return a list of invalidation projects with their invalidation reasons
+         * @since 8.10
+         */
+        List<ProjectInvalidationReasons> getProjectInvalidationReasons();
+    }
+
+    /**
+     * Contains data of a single invalidated project.
+     *
+     * @since 8.10
+     */
+    public interface ProjectInvalidationReasons {
+        /**
+         * Returns the identity path of the invalidated project.
+         *
+         * @return the identity path
+         * @since 8.10
+         */
+        String getIdentityPath();
+
+        /**
+         * Returns the list of invalidation reasons. At least one invalidation is always present.
+         *
+         * @return the list of invalidation reasons
+         * @since 8.10
+         */
+        List<FingerprintInvalidationReason> getInvalidationReasons();
+    }
+
+    /**
+     * Represents the invalidation reason.
+     *
+     * @since 8.10
+     */
+    public interface FingerprintInvalidationReason {
+        /**
+         * Returns a simple text representation of the invalidation reason.
+         *
+         * @return the invalidation reason as text
+         * @since 8.10
+         */
+        String getMessage();
+    }
+
+    /**
+     * An overall result of the fingerprint check.
+     */
+    public enum CheckStatus {
```

Comment:
❔ What status can we have when running tasks? My understanding is that we'll only have `NOT_FOUND`, `VALID`, and `INVALID`, right? And the partial status is only when requesting build models.
How might that change in the future?

Reply:
At some point, IP would be able to configure projects incrementally for all scenarios (running tasks and/or creating models). So, I would assume all statuses are applicable regardless of whether tasks were requested.

I would also assume that sometime in the future we'll introduce support for CC when creating models (regardless whether tasks are running or not). Currently, CC is disabled whenever models are created.

Reply:
> ❔ What status can we have when running tasks? My understanding is that we'll only have `NOT_FOUND`, `VALID`, and `INVALID`, right? And the partial status is only when requesting build models. How might that change in the future?

As Alex said, we might be able to have `PARTIAL` hits for task executions too, but this is a future (and somewhat distant) work.